src/azure-cli-core/azure/cli/core/commands/client_factory.py

```diff
@@ -20,16 +20,19 @@
 
 
 def get_mgmt_service_client(client_or_resource_type, subscription_id=None, api_version=None,
-                            **kwargs):
+                            sdk_profile=None, **kwargs):
     if isinstance(client_or_resource_type, ResourceType):
         # Get the versioned client
         client_type = get_client_class(client_or_resource_type)
         api_version = api_version or get_api_version(client_or_resource_type)
+        if isinstance(api_version, SDKProfile):
```

Comment:
should we treat it as an error when both `api_version` and `sdk_profile` are set?

Reply:
No, it's not opposite, since api_version is the default if profile does not contain a mapping for a specific RT